### PR TITLE
Rwa 3193 mandatory fileds add validation during initiation

### PR DESCRIPTION
**Pull request by skandala-cgi**
### JIRA link (if applicable) ###
https://tools.hmcts.net/jira/browse/RWA-3940

### Change description ###
mandatory fileds add validation during initiation

**Does this PR introduce a breaking change?** (check one with "x")

```
[x] Yes
[ ] No
```

### CODE REVIEW

File: src/integrationTest/java/uk/gov/hmcts/reform/wataskmanagementapi/utils/TaskMandatoryFieldsValidatorIntegrationTest.java [hunk 1]
```diff
@@ -53,19 +57,57 @@ void setUp() {
...
+        "'someTaskName', 'someTaskType', 'PUBLIC', 'title', 'CASE_ID', 'Asylum', 'CaseCategory', 'CaseName', 'IA', 'TestRegion', '765324', 'roleCategory', ",
+
+    })
+    @DisplayName("should throw ServiceMandatoryFieldValidationException when a mandatory field is missing")
+    void should_throw_service_mandatory_field_validation_exception_when_a_mandatory_field_is_missing(String taskName, String taskType, String securityClassification, String title,  String caseId, String caseTypeId, String caseCategory, String caseName, String jurisdiction, String region, String location, String roleCategory, String workTypeResource) {
+        TaskResource task = getTaskResource(taskId, taskName, taskType, securityClassification, title, caseId, caseTypeId, caseCategory, caseName, jurisdiction, region, location, roleCategory, workTypeResource);
+        ServiceMandatoryFieldValidationException exception = assertThrows(ServiceMandatoryFieldValidationException.class, () -> taskMandatoryFieldsValidator.validate(task));
+        String message = exception.getMessage();
+        assertTrue(message.contains("cannot be null or empty"));
```
Comment (raghera) [inline src/integrationTest/java/uk/gov/hmcts/reform/wataskmanagementapi/utils/TaskMandatoryFieldsValidatorIntegrationTest.java:93]: could you improve this to include the name of the field that cannot be null or empty?

File: src/functionalTest/java/uk/gov/hmcts/reform/wataskmanagementapi/watasks/controllers/PostTaskInitiateByIdControllerTest.java [hunk 2]
```diff
@@ -873,6 +875,37 @@ public void should_return_a_503_if_task_already_initiated_however_handled_gracef
         common.cleanUpTask(taskId);
     }
 
+    @Test
+    public void should_return_a_502_if_task_is_missing_mandatory_task_attributes() {
```
[the remark this answers is not in the document]
Comment (raghera) [reply src/functionalTest/java/uk/gov/hmcts/reform/wataskmanagementapi/watasks/controllers/PostTaskInitiateByIdControllerTest.java:880]: Has anything changed regarding this one?

[the remark this answers is not in the document]
Comment (raghera) [reply src/functionalTest/java/uk/gov/hmcts/reform/wataskmanagementapi/watasks/controllers/PostTaskInitiateByIdControllerTest.java:880]: This Jira is related: https://tools.hmcts.net/jira/browse/RWA-3970

File: src/testUtils/java/uk/gov/hmcts/reform/wataskmanagementapi/SpringBootFunctionalBaseTest.java [hunk 3]
```diff
@@ -296,7 +299,24 @@ private Consumer<Response> defaultInitiationAssert(TestVariables testVariables)
...
+        Response response = restApiActions.post(
+            TASK_INITIATION_ENDPOINT,
+            testVariables.getTaskId(),
+            initiateTaskRequest,
+            authorizationProvider.getServiceAuthorizationHeadersOnly()
+        );
+
+        //Note: Since tasks can be initiated directly by task monitor, we will have database conflicts for
+        // second initiation request, so we are by-passing 503 and 201 response statuses.
+        assertResponse(response);
```
[the remark this answers is not in the document]
Comment (skandala-cgi) [reply src/testUtils/java/uk/gov/hmcts/reform/wataskmanagementapi/SpringBootFunctionalBaseTest.java:314]: Raised https://tools.hmcts.net/jira/browse/RWA-4020

**Review by raghera — DISMISSED** [review]

File: src/main/java/uk/gov/hmcts/reform/wataskmanagementapi/services/utils/TaskMandatoryFieldsValidator.java [hunk 4]
```diff
@@ -135,7 +141,19 @@ public void validateTaskMandatoryFields(TaskResource task) {
...
+     * Checks if the given field value is null or empty.
+     *
+     * @param fieldValue the value of the field to be checked
+     * @return true if the field value is null or empty, false otherwise
+     */
     private boolean isFieldNullOrEmpty(Object fieldValue) {
+        if (fieldValue instanceof WorkTypeResource workTypeResource) {
+            return workTypeResource.getId() == null || workTypeResource.getId().isBlank();
+        } else if (fieldValue instanceof ExecutionTypeResource executionTypeResource) {
+            return executionTypeResource.getExecutionCode() == null
```
Comment (BenNzewi) [inline src/main/java/uk/gov/hmcts/reform/wataskmanagementapi/services/utils/TaskMandatoryFieldsValidator.java:154]: test coverage?

**Review by BenNzewi — APPROVED** [review]

**Pull request merged**